lib/main.js

```diff
@@ -61,10 +64,27 @@ var E = '',
 lift = function (obj, namespace, separator, transformer, _debug) {
     var key,
         cast,
-        eKey;
+        eKey,
+        allKeys,
+        keyCount;
 
     // iterate on each item in this object and operate on their keys.
-    for (key in obj) {
+    if (Array.isArray(obj)) {
+        allKeys = Array.apply(null, {
+            length: obj.length
+        }).map(Number.call, Number);
+    }
+    else if (typeof obj === 'object') {
+        allKeys = Object.keys(obj);
+    }
+    else {
+        // Base case, where the object is a primitive
```

Comment:
The return value of the `lift` function is not utilised anywhere and as such, gave me quite a chase to understand the purpose of this... sadly I found none. :-/

Should this be `return obj` or just `return`?